tests/unit/amo/pages/TestUserProfile.js

```diff
@@ -102,488 +94,438 @@ describe(__filename, () => {
-    expect(root.find(AddonsByAuthorsCard).at(1)).toHaveProp(
-      'authorDisplayName',
-      null,
-    );
+    expect(screen.queryAllByClassName('AddonsByAuthorsCard')).toHaveLength(2);
   });
 
   it('renders AddonsByAuthorsCard for extensions', () => {
     const userId = 123;
-    const { params, store } = signInUserWithUserId(userId);
+    const { params } = signInUserWithProps({ userId });
+    const user = getCurrentUser(store.getState().users);
 
-    const root = renderUserProfile({ params, store });
+    renderUserProfile({ params });
 
-    expect(root.find(AddonsByAuthorsCard).at(0)).toHaveProp(
-      'addonType',
-      ADDON_TYPE_EXTENSION,
-    );
-    expect(root.find(AddonsByAuthorsCard).at(0)).toHaveProp(
-      'pageParam',
-      'page_e',
-    );
-    expect(root.find(AddonsByAuthorsCard).at(0)).toHaveProp('paginate', true);
-    expect(root.find(AddonsByAuthorsCard).at(0)).toHaveProp(
-      'pathname',
-      `/user/${userId}/`,
-    );
+    expect(screen.getByText(`Extensions by ${user.name}`)).toBeInTheDocument();
+    const extensionCard = screen
+      .queryAllByClassName('AddonsByAuthorsCard')
+      .item(0);
+    expect(
+      within(extensionCard).queryByText('Previous'),
+    ).not.toBeInTheDocument();
+    expect(within(extensionCard).queryByText('Next')).not.toBeInTheDocument();
   });
 
   it('renders AddonsByAuthorsCard for themes', () => {
     const userId = 123;
-    const { params, store } = signInUserWithUserId(userId);
+    const { params } = signInUserWithProps({ userId });
+    const user = getCurrentUser(store.getState().users);
 
-    const root = renderUserProfile({ params, store });
+    renderUserProfile({ params });
 
-    expect(root.find(AddonsByAuthorsCard).at(1)).toHaveProp(
-      'addonType',
-      ADDON_TYPE_STATIC_THEME,
-    );
-    expect(root.find(AddonsByAuthorsCard).at(1)).toHaveProp(
-      'pageParam',
-      'page_t',
-    );
-    expect(root.find(AddonsByAuthorsCard).at(1)).toHaveProp('paginate', true);
-    expect(root.find(AddonsByAuthorsCard).at(1)).toHaveProp(
-      'pathname',
-      `/user/${userId}/`,
-    );
+    expect(screen.getByText(`Themes by ${user.name}`)).toBeInTheDocument();
+    const themeCard = screen.queryAllByClassName('AddonsByAuthorsCard').item(1);
+    expect(screen.queryByText(themeCard, 'Previous')).not.toBeInTheDocument();
+    expect(screen.queryByText(themeCard, 'Next')).not.toBeInTheDocument();
   });
 
-  it('renders a not found page if the API request is a 404', () => {
-    const { store } = dispatchSignInActions();
-    const errorHandler = new ErrorHandler({
-      id: 'some-error-handler-id',
-      dispatch: store.dispatch,
-    });
-    errorHandler.handle(
-      createApiError({
-        response: { status: 404 },
-        apiURL: 'https://some/api/endpoint',
-        jsonResponse: { message: 'not found' },
-      }),
-    );
+  // it('renders a not found page if the API request is a 404', () => {
```

Comment:
This test isn't working - the not found message doesn't seem to be rendered - and I think it's got something to do with the `ErrorHandler` not being used.  (similar-ish seemingly `ErrorHandler` related problems in other tests)

Reply:
I created a new helper function called `createFailedErrorHandler`, and all you have to do is call it and it will record the information about the error in the store. The important bit is to call it with the error you want stored, and the proper `id`, which needs to match the id that would be generated for your component's errorHandler. You already have a function to generate that `id` in your test, called `createErrorHandlerId`.

Here is a diff including both the new function, and changes to one of your tests that gets it passing. You should be able to use this technique for any tests that need errors to have been handled:
```diff
diff --git a/tests/unit/amo/pages/TestUserProfile.js b/tests/unit/amo/pages/TestUserProfile.js
index 482831bf7..68d8da2aa 100644
--- a/tests/unit/amo/pages/TestUserProfile.js
+++ b/tests/unit/amo/pages/TestUserProfile.js
@@ -13,12 +13,12 @@ import {
   loadUserAccount,
   FETCH_USER_ACCOUNT,
 } from 'amo/reducers/users';
-import { DEFAULT_API_PAGE_SIZE } from 'amo/api';
-// import { DEFAULT_API_PAGE_SIZE, createApiError } from 'amo/api';
+import { DEFAULT_API_PAGE_SIZE, createApiError } from 'amo/api';
 import { CLIENT_APP_FIREFOX, USERS_EDIT } from 'amo/constants';
 // import { ErrorHandler } from 'amo/errorHandler';
 import { sendServerRedirect } from 'amo/reducers/redirectTo';
 import {
+  createFailedErrorHandler,
   createHistory,
   // createStubErrorHandler,
   createUserAccountResponse,
@@ -471,26 +471,24 @@ describe(__filename, () => {
     expect(screen.queryByText(themeCard, 'Next')).not.toBeInTheDocument();
   });

-  // it('renders a not found page if the API request is a 404', () => {
-  //   const errorHandler = new ErrorHandler({
-  //     id: 'some-error-handler-id',
-  //     dispatch: store.dispatch,
-  //   });
-  //   errorHandler.handle(
-  //     createApiError({
-  //       response: { status: 404 },
-  //       apiURL: 'https://some/api/endpoint',
-  //       jsonResponse: { message: 'not found' },
-  //     }),
-  //   );
+  it('renders a not found page if the API request is a 404', () => {
+    const userId = 123;
+    createFailedErrorHandler({
+      error: createApiError({
+        response: { status: 404 },
+        apiURL: 'https://some/api/endpoint',
+        jsonResponse: { message: 'not found' },
+      }),
+      id: createErrorHandlerId({ userId }),
+      store,
+    });

-  //   renderUserProfile({ errorHandler });
+    renderUserProfile({ params: { userId } });

-  //   screen.debug(screen.getByClassName('Page-content'));
-  //   expect(
-  //     screen.getByText('Oops! We can’t find that page'),
-  //   ).toBeInTheDocument();
-  // });
+    expect(
+      screen.getByText('Oops! We can’t find that page'),
+    ).toBeInTheDocument();
+  });

   // it('renders errors', () => {
   //   const errorHandler = new ErrorHandler({
diff --git a/tests/unit/helpers.js b/tests/unit/helpers.js
index 0b27da559..d5184f798 100644
--- a/tests/unit/helpers.js
+++ b/tests/unit/helpers.js
@@ -1655,3 +1655,19 @@ export const renderPage = (ui, options = {}) => {
   // render the correct page.
   return render(<App>{ui}</App>, options);
 };
+
+export const createFailedErrorHandler = ({
+  error = new Error(),
+  id = 'some-error-handler-id',
+  message = 'An error message',
+  store,
+}) => {
+  invariant(store, 'store must be passed into createFailedErrorHandler');
+  const errorHandler = new ErrorHandler({
+    dispatch: store.dispatch,
+    id,
+  });
+  errorHandler.handle(error);
+  errorHandler.addMessage(message);
+  return errorHandler;
+};
```